pkg/cmd/experimental/diagnostics/client.go

```diff
@@ -34,7 +34,7 @@ func (o DiagnosticsOptions) buildClientDiagnostics(rawConfig *clientcmdapi.Confi
 		switch diagnosticName {
 		case clientdiags.ConfigContextsName:
 			for contextName := range rawConfig.Contexts {
-				diagnostics = append(diagnostics, clientdiags.ConfigContext{rawConfig, contextName})
+				diagnostics = append(diagnostics, clientdiags.ConfigContext{RawConfig: rawConfig, ContextName: contextName})
```

Comment:
The safety it gives us in case a struct's definition changes is well worth the effort, in my opinion. If you don't like naming fields, make them private and expose a constructor `New()` method.